pkg/services/dashboards/service/dashboard_service_test.go

```diff
@@ -54,224 +54,101 @@ func TestMain(m *testing.M) {
-				dto.User = &user.SignedInUser{UserID: 1}
-				_, err := service.SaveProvisionedDashboard(context.Background(), dto, nil)
-				require.NoError(t, err)
-			})
+		t.Run("When saving a dashboard with empty title it should return error", func(t *testing.T) {
+			titles := []string{"", " ", "   \t   "}
 
-			t.Run("Should override invalid refresh interval if dashboard is provisioned", func(t *testing.T) {
-				fakeStore.On("SaveProvisionedDashboard", mock.Anything, mock.AnythingOfType("dashboards.SaveDashboardCommand"), mock.AnythingOfType("*dashboards.DashboardProvisioning")).Return(&dashboards.Dashboard{Data: simplejson.New()}, nil).Once()
-				oldRefreshInterval := service.cfg.MinRefreshInterval
-				service.cfg.MinRefreshInterval = "5m"
-				defer func() { service.cfg.MinRefreshInterval = oldRefreshInterval }()
-
-				dto.Dashboard = dashboards.NewDashboard("Dash")
-				dto.Dashboard.SetID(3)
-				dto.User = &user.SignedInUser{UserID: 1}
-				dto.Dashboard.Data.Set("refresh", "1s")
-				_, err := service.SaveProvisionedDashboard(context.Background(), dto, nil)
-				require.NoError(t, err)
-				require.Equal(t, dto.Dashboard.Data.Get("refresh").MustString(), "5m")
-			})
+			for _, title := range titles {
+				dto.Dashboard = dashboards.NewDashboard(title)
+				_, err := service.SaveDashboard(ctx, dto, false)
+				require.Equal(t, err, dashboards.ErrDashboardTitleEmpty)
+			}
 		})
 
-		t.Run("Import dashboard validation", func(t *testing.T) {
-			dto := &dashboards.SaveDashboardDTO{}
-
-			t.Run("Should return validation error if dashboard is provisioned", func(t *testing.T) {
-				fakeStore.On("GetProvisionedDataByDashboardID", mock.Anything, mock.AnythingOfType("int64")).Return(&dashboards.DashboardProvisioningSearchResults{}, nil).Once()
-
-				dto.Dashboard = dashboards.NewDashboard("Dash")
-				dto.Dashboard.SetID(3)
-				dto.User = &user.SignedInUser{UserID: 1}
-				_, err := service.ImportDashboard(context.Background(), dto)
-				require.Equal(t, err, dashboards.ErrDashboardCannotSaveProvisionedDashboard)
-			})
+		t.Run("Should return validation error if message is too long", func(t *testing.T) {
+			dto.Dashboard = dashboards.NewDashboard("Dash")
+			dto.Message = `Here we go, 500+ characters for testing. I'm sorry that you're
+			having to read this. I spent too long trying to come up with something clever
+			to say or a funny joke. Unforuntately, nothing came to mind. So instead, I'm
+			will share this with you, as a form of payment for having to read this:
+			https://youtu.be/dQw4w9WgXcQ?si=KeoTIpn9tUtQnOBk! Enjoy :) Now lets see if
+			this test passes or if the result is more exciting than these 500 characters
+			I wrote. Best of luck to the both of us!!`
+			_, err := service.SaveDashboard(ctx, dto, false)
+			require.Equal(t, err, dashboards.ErrDashboardMessageTooLong)
+
+			// set to a shorter message for the rest of the tests
+			dto.Message = `message`
 		})
 
-		t.Run("Given provisioned dashboard", func(t *testing.T) {
```

Comment:
dashboard provisioning validation is tested here: https://github.com/grafana/grafana/blob/main/pkg/registry/apis/dashboard/register_test.go#L24 and in pkg/tests/apis/dashboard